packages/gatsby/src/utils/flags.ts

```diff
@@ -235,6 +235,15 @@ const activeFlags: Array<IFlag> = [
     experimental: false,
     testFitness: (): fitnessEnum => true,
   },
+  {
+    name: `GRAPHQL_TYPEGEN`,
+    env: `GATSBY_GRAPHQL_TYPEGEN`,
+    command: `develop`,
+    telemetryId: `GraphQLTypegen`,
+    description: `A built-in way for automatic TypeScript type generation and better GraphQL IntelliSense. See https://github.com/gatsbyjs/gatsby/discussions/35420 for more details.`,
+    experimental: true,
```

Comment:
I'm not 100% sure on this:

```suggestion
    experimental: true,
    noCI: true,
```

So treat this mostly as a question wether we should run typegen in CI or not

Reply:
I added it for now -- if people want that we could change it